### ISSUE TITLE
heap profiling and realloc

### ISSUE THREAD

**Issue by mmimica** [jemalloc/jemalloc#557]
Heap profiling with [realloc](https://linux.die.net/man/3/realloc) doesn't seem to work right.
```
#include <stdlib.h>
int main() {
	int *p = malloc(1024);
	p = realloc(p, 2048);
	free(p);	
	return 0;
}
```
`$ MALLOC_CONF="prof:true,lg_prof_sample:0,lg_prof_interval:0,prof_final:true" ./testapp`
`$ jeprof --show_bytes testapp jeprof.29410.7.f.heap`

It reports a leak (live blocks):
(jeprof) top
Total: 1024 B
    1024 100.0% 100.0%     1024 100.0% main
       0   0.0% 100.0%     1024 100.0% __libc_start_main
       0   0.0% 100.0%     1024 100.0% _start

If I remove the "realloc" line there is no reported leak. I'm I doing something wrong here?

**Comment by jasone** [comment]
Does this look to you like the same problem as #499?.

**Comment by mmimica** [comment]
Exactly. Thanks!

**Comment by mmimica** [comment]
I produced this with 4.4.0. Do you know which is the last version that has this feature working?

**Comment by jasone** [comment]
Oh, I thought this was a dev-only issue!  If it affects 4.x, I don't know how far back the problem goes, but we certainly want to be sure 4.5 doesn't perpetuate the problem.  I'm going to set the milestone to 4.5 to make sure we investigate in time.

**Comment by mmimica** [comment]
No, sorry, it works fine in 4.4.0. I was unaware I was using master.

**Comment by jasone** [comment]
I verified that this reproduces on the dev branch, but not the master (release) branch.  Closing as a duplicate of #499, which we'll fix for 5.0.0.
